subprojects/language-native/src/main/java/org/gradle/language/cpp/plugins/CppLibraryPlugin.java

```diff
@@ -96,6 +98,18 @@ public void apply(final ProjectInternal project) {
         // TODO - add lifecycle tasks
         // TODO - extract some common code to setup the configurations
 
+        Provider<RegularFile> moduleMapFile = project.getLayout().getBuildDirectory().file("map/module.modulemap");
+        GenerateModuleMap generateModuleMap = tasks.create("generateModuleMap", GenerateModuleMap.class);
+        generateModuleMap.getModuleMapFile().set(moduleMapFile);
+        generateModuleMap.getModuleName().set(library.getBaseName());
+        generateModuleMap.getPublicHeaderDirs().addAll(project.provider(new Callable<Iterable<File>>() {
+            @Override
+            public Iterable<File> call() throws Exception {
+                return library.getPublicHeaderDirs();
+            }
+        }));
+        library.getModuleMapFile().set(generateModuleMap.getModuleMapFile());
```

Comment:
Hmmm...not sure.  I think this is just the tip of a broader question which we've already touched on in other conversations which is "should the generation of the module map be associated with the producer or the consumer".  Right now it's associated with the producer, but there are some good arguments for associating it with the consumer, in which case I think that yes, this probably should only be enabled for a clang toolchain.  On the other hand, if it's associated with the producer, it's less clear - i.e. why should I not create/publish a module map just because I happen to be building the binary with a non-clang toolchain.  Given that these two things are coupled right now, though, it's probably safe to just say only do this if the toolchain supports module maps, but I can see this coming back up again in the future.

Reply:
Everything has been moved to the consumer side now, so this is no longer a thing.